src/Games/NexusMods.Games.MountAndBlade2Bannerlord/LauncherManager/LauncherManagerNexusMods.cs

```diff
@@ -0,0 +1,61 @@
+using Bannerlord.LauncherManager;
+using Bannerlord.LauncherManager.External;
+using Bannerlord.LauncherManager.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.Settings;
+
+namespace NexusMods.Games.MountAndBlade2Bannerlord.LauncherManager;
+
+public sealed partial class LauncherManagerNexusMods : LauncherManagerHandler,
```

Comment:
There's a lot of layers of logic here. Namely, this inherits from `LauncherManagerHandler` which itself consists of many partial classes, segregated by trait-like features.

There needs to be XML Documentation (and notes) of some sort, either here or in the `Bannerlord.LauncherManager` dependency, as for most (almost all?) readers the purpose of this code will be unclear. Especially if they haven't dug into Bannerlord previously.

Just note that this is the `Adapter` class, so to speak; which translates/marshals internal library logic and exposes them via NMA primitives. Some minimal info should be provided on the methods.

- `LauncherState` (`GetState`) seems to toggle between SP/MP, which we may need to inject some sort of button in the UI for.
- `LoadOrder` (`LoadLoadOrder/SaveLoadOrder`) serializes/deserializes load order as a dictionary. Reading in from Bannerlord, and saving from App back to Bannerlord. Every entry here has the index as part of its class layout, so changing one requires changing all other elements.
- `IModuleViewModel` (`GetAllModuleViewModels/GetModuleViewModels/SetModuleViewModels`) is for reading/writing loadouts (no load order).

Some minimal info on how these systems interact with NMA would be needed.
i.e. something along the lines of `Imports the module setup from the game folder to update NMA's internal state`

-----------------

In the long term, I imagine we'd want to use `LauncherManagerNexusMods` to 'ingest' (integrate, import) changes user made in the game's native launcher. Although we've talked about this topic as a  team last week, the result wasn't fully conclusive.

I do think though that using `Bannerlord.LauncherManager` and current disk state as the source of truth would be preferable, so if user rearranges mods in native launcher, we'd apply that; but the mechanisms around that aren't fully set in stone.